### DISCUSSION TITLE
Polinovel

### DISCUSSION THREAD

**Discussion by seamaster101** [Louisvdw/dbus-serialbattery#486] [New BMS requests]
Hi all,
My batteries are from manufacturer called Polinovel. I know that the they are 300Ahrs and have 150A BMSs with RS-485 communications.
It uses the protocol attached in a PDF file. I would like to make them work with dbus-serialbattery. The batteries  manufacturer did not tell me what make and model the BMSs are, but at least provided the communication protocol. 
Can someone recognize the protocol and tell me what type is the BMS they used tin my batteries?    
[BMS-RS232-RS485-protocol-20191108.pdf](https://github.com/Louisvdw/dbus-serialbattery/files/10928257/BMS-RS232-RS485-protocol-20191108.pdf)
I also started a driver, but honestly programming is not my forte. The first part of the driver works, and I'm able to query the BMS and the BMS sends data back, but from there I have problems parsing the voltages and other parameters. 

The polinovel.py.ziip is the file I started, but can not make work.
[polinovell.py.zip](https://github.com/Louisvdw/dbus-serialbattery/files/10928265/polinovell.py.zip)
Please help!

**Comment by seamaster101** [comment]
Bump

**Comment by mr-manuel** [comment]
1. So what is the current state of this?
2. What are the open problems? 
3. What values are you successfully reading?
4. What es the answer that the BMS report back?

Please fork this repository and add your script in your forked repository. This would be easier to see your progress and help you with development. If you have no programming editor yet, I can recommend you to use [VS Code](https://code.visualstudio.com/download).

Once the driver is working, you can open a pull request to merge it with the `dev` branch and make it available for everyone.

**Reply by seamaster101** [reply]
I read the batteries now but it can not be easily applied to anyone else even with the same batteries. For ones, I don't know what exactly BMS they have inside. I know part of the protocol that was given to me by the battery manufacturer, but it is being read by additional  hardware that I made to read each battery RS485 output and create that page.  So not sure if will help anyone as no one else will have this web page that I'm parsing and sending the data to bus-mqtt-battery. what do you think?

**Reply by mr-manuel** [reply]
I'm not talking about your webpage, but the RS485 driver that reads directly from the BMS. So you are fine with your actual solution (`Battery` -> `ESP` -> `Node-RED` -> `dbus-mqtt-battery`)?

If yes, then we can close this battery request.

**Comment by seamaster101** [comment]
The 485 driver does not run on the RPI. It does run on a piece of custom hardware, therefore will not work for anyone else as they don’t have that hardware. I did start trying to make driver from Louise's code, but I got stuck :(. I got it up to the point where it pareces one value. I don’t know if that file could be of assistance   My programming skills are not sufficient to finish the driver on my own. Jordan

**Reply by mr-manuel** [reply]
Which value is parsed? More stable would surely be the direct connection without going over three additional steps. Please post the code here, even if it's a other programming language. The more informations you give, the better we can help.
